File: Exercicios/Java/Lista1/src/Ex6.java
import java.io.IOException;

/**
 * SSC0103 - Programação Orientada a Objetos
 * Lista de exercícios #1 (Java)
 * Exercício 06
 * 
 * @author Ana Lívia Ruegger Saldanha (N.USP 8586691)
 *
 */
public class Ex6 {
	
	/**
	 * Este método lê um número inteiro e exibe o primeiro número
	 * primo menor que o número informado.
	 * 
	 * @param args
	 * @throws IOException
	 */
	public static void main(String[] args) throws IOException {
		int n = EntradaTeclado.leInt();
		for (int i = n - 1; i >= 2; i--) {
			boolean i_is_prime = true;
			for (int j = 2; j <= i/2; j++) {
				if (i % j == 0) {
					i_is_prime = false;
					break;
				}
			}
			if (i_is_prime == true) {
				System.out.print(i);
				return;
			}
		}
		System.out.printf("Não existe número primo menor que %d.", n);
	}

}

File: Exercicios/Python/Lista1/Ex6.py
# SSC0103 - Programação Orientada a Objetos
# Lista de exercícios #1 (Python)
# Exercício 06
# 
# Ana Lívia Ruegger Saldanha (N.USP 8586691)

'''
Este programa lê um número inteiro e exibe o primeiro número primo menor
que o número informado.
'''


def nearest_prime(n):
    for i in range(n - 1, 1, -1):
        i_is_prime = True
        for j in range(2, int(i/2 + 1)):
            if i % j == 0:
                i_is_prime = False
                break
        
        if i_is_prime == True:
            print(i)
            return

    print("Não existe número primo menor que", n)

n = int(input())
nearest_prime(n)